tests/integration_tests.rs

```diff
@@ -209,21 +211,50 @@ fn line_range_multiple() {
         .stdout("line 1\nline 2\nline 4\n");
 }
 
+fn setup_cycle_file(filename: &str, content: &[u8]) -> PathBuf {
+    let path = PathBuf::from(EXAMPLES_DIR).join(filename);
+    let _ = File::create(path.as_path())
+        .expect("Could not open file for writing")
+        .write_all(content)
+        .expect("Failure during write");
+    path
+}
+
 #[test]
 fn basic_io_cycle() {
-    let file_out = Stdio::from(File::open("tests/examples/cycle.txt").unwrap());
+    let filename = setup_cycle_file("cycle1.txt", b"I am not empty");
```

Comment:
This file is being created, but it's also checked in to the repository. Is that intended?

We usually create temporary directories if we really want to modify / create files for tests.

Reply:
I was gonna ask you about replacing `tempdir` with `tempfile` but just saw that it is already switched :smile: Yep, that is cleaner

Reply:
Maybe that would resolve the permission problems on Windows as well?

Reply:
I read the documentation of `tempfile` and since a temporary file would only give me the file and not a path to pass to `bat`, I ended up using directories after all. However, this did not fix the problems on Windows. I have actually read over the code again and found that another test shouldn't fail on Windows, so I added that in and obviously, that one also fails with a permission error. I have to do some more reading.

I first thought that the problem occurs, because temporary files may never be written to disk and the subprocess wouldn't be able to access them, but a `TempDir` just wraps a path to a normal directory (https://github.com/Stebalien/tempfile/blob/master/src/dir.rs#L403). It wouldn't explain why the tests failed with the files in `tests/examples` either.

This might take some more time :man_shrugging:

Reply:
After some intense sessions in the stdlib source code and msdn, I decided to add some `dbg!` stuff in my windows vm and to my surprise, it seemed that not the tempfile is the problem here, but `test.txt` (spoiler: it something completely different):

![image](https://user-images.githubusercontent.com/47574893/109216841-598b0100-77b5-11eb-8ef6-e7c8080f5d02.png)

(the thing to see here is that the error is printed before the `index = 1` which is the index in the main loop in `controller.rs`, so the error happens before the `-` argument or the temp file are even opened :scream: )

So I checked the security tab for `test.txt` and compared it to `tabs.txt`, but they both had the exact same list, so that can't it. I guess the access denied has nothing todo with reading after all ...

---

I narrowed it down to somewhere in `print_file`, so the access denied happens there somewhere. I will see what happens under hood when the process stdout gets pointed to a file and what sharing options are given and what not

Reply:
I found another small mistake, but I does not seem to be the problem. We could also consider making these tests unix exclusive. After all, we are _not doing anything_ on windows anyways. What do you think?

Reply:
Thank you for looking into this.

Running these tests on unix only is what I suggested below, right? I'd be okay with that. I'd also be okay with completely disabling `clircle` functionality on Windows.